src/util_exec.cpp

```diff
@@ -579,7 +579,7 @@ Util::runMtcpRestore(int is32bitElf,
     memset(dummyEnviron, '0', argvSizeDiff - 1);
     strncpy(dummyEnviron,
             ENV_VAR_DMTCP_DUMMY "=",
-            strlen(ENV_VAR_DMTCP_DUMMY "="));
+            argvSizeDiff);
```

Comment:
We just want to copy the `"DMTCP_DUMMY_ENV="` string to the beginning of `dummyEnviron`. That way, the buffer `dummyEnviron` will contain `"DMTCP_DUMMY_ENV=00000...0"`

Reply:
I agree with @karya0.  We need a better fix here.